### PR TITLE
Make iOS RN metadata schema match docs

### PR DESCRIPTION
**Pull request by pladaub**

### CODE REVIEW

File: ios/RNLinksdk.m [hunk 1]
```diff
@@ -576,26 +577,196 @@ + (NSString *)subtypeNameForAccountSubtype:(id<PLKAccountSubtype>)accountSubtype
...
++ (NSString *)errorMessageFromError:(PLKExitError *)error {
+    return error.userInfo[kPLKExitErrorMessageKey] ?: @"";
+}
+
++ (NSString *)stringForEventName:(PLKEventName *)eventName {
+    if (!eventName) {
+        return @"";
+    }
+    
+    if (eventName.unknownStringValue) {
```
Comment (amytang0) [inline ios/RNLinksdk.m:733]: `&& [eventName.unknownStringValue length]` ?
Since unknownStringValue can be empty?

Comment (pladaub) [reply ios/RNLinksdk.m:733]: Hrm, this is a little tricky – if the iOS SDK returns .unknown("") we probably do want to have this result in @"" being returned here.

**Review by amytang0 — DISMISSED** [review]
Remember to fix conflicts and test before merging

LGTM, some food for thought, Android is converting using a JsonFormatter, which nullifies the need for these `toXString` methods. Not sure if that strategy is viable with the Obj-C enums

File: ios/RNLinksdk.m [hunk 2]
```diff
@@ -577,26 +579,195 @@ + (NSString *)subtypeNameForAccountSubtype:(id<PLKAccountSubtype>)accountSubtype
...
+    return @{
+        @"event": [self stringForEventName:event.eventName] ?: @"",
+        @"metadata": @{
+            @"error_type": [self errorTypeStringFromError:metadata.error] ?: @"",
+            @"error_code": [self errorCodeStringFromError:metadata.error] ?: @"",
+            @"error_message": [self errorMessageFromError:metadata.error] ?: @"",
+            @"exit_status": [self stringForExitStatus:metadata.exitStatus] ?: @"",
+            @"institution_id": metadata.institutionID ?: @"",
+            @"institution_name": metadata.institutionName ?: @"",
+            @"instituion_search_query": metadata.institutionSearchQuery ?: @"",
```
Comment (amytang0) [inline ios/RNLinksdk.m:607]:
```suggestion
            @"institution_search_query": metadata.institutionSearchQuery ?: @"",
```

**Review by amytang0 — APPROVED** [review]
LGTM

**Pull request merged**